openmdao/recorders/sqlite_reader.py

```diff
@@ -402,7 +435,15 @@ def list_cases(self, source=None, recurse=True, flat=True):
                             (source, type(source).__name__))
 
         if not source:
-            return self._list_cases_recurse_flat()
+            if not out_stream:
+                return self._list_cases_recurse_flat()
+            else:
+                cases = self._list_cases_recurse_flat()
+                if out_stream is _DEFAULT_OUT_STREAM:
+                    out_stream = sys.stdout
+
+                if out_stream:
+                    return write_source_table(self.source_cases_table, out_stream)
```

Comment:
I think this should always have a return value (`cases`) regardless of the out_stream, similar to the list_ functions on System (and in keeping with the docstring)
The `write_source_table` function does not return anything.

Reply:
Seems like it would be cleaner to just pass the `out_stream` arg to `_list_cases_recurse_flat`, then you wouldn't have to rely on the private attribute/side effect and this extra logic here...